driver-mapping/src/main/java/com/datastax/driver/mapping/AnnotationChecks.java

```diff
@@ -40,7 +40,7 @@
     static <T extends Annotation> T getTypeAnnotation(Class<T> annotation, Class<?> annotatedClass) {
         T instance = annotatedClass.getAnnotation(annotation);
         if (instance == null)
-            throw new IllegalArgumentException(String.format("@%s annotation was not found on %s",
+            throw new IllegalStateException(String.format("@%s annotation was not found on %s",
```

Comment:
Admittedly `IllegalArgumentException` isn't the most appropriate type for mapper annotation misuse, but `IllegalStateException` isn't either: 
* `IllegalArgumentException`: *Thrown to indicate that a method has been passed an illegal or inappropriate argument*
* `IllegalStateException`: *Signals that a method has been invoked at an illegal or inappropriate time*

Between the two, I'd rather keep IAE for the sake of backward compatibility.